rustworkx-core/src/connectivity/minimum_cycle_basis.rs

```diff
@@ -0,0 +1,493 @@
+        let new_node = gi.add_node(gi_node_name.clone());
+        let new_node_index = gi.to_index(new_node);
+        let lifted_node = gi.add_node(gi_lifted_node_name.clone());
+        let lifted_node_index = gi.to_index(lifted_node);
+        gi_name_to_node_index.insert(gi_node_name, new_node_index);
+        gi_name_to_node_index.insert(gi_lifted_node_name, lifted_node_index);
+    }
+    // # Add 2 copies of each edge in G to Gi.
+    // # If edge is in orth, add cross edge; otherwise in-plane edge
+    for edge in graph.edge_references() {
+        let u_index = graph.to_index(edge.source());
+        let v_index = graph.to_index(edge.target());
+        let u_name = format!("{:?}", graph.node_weight(edge.source()).unwrap())
+            .trim_matches('"')
+            .to_string();
+        let v_name = format!("{:?}", graph.node_weight(edge.target()).unwrap())
+            .trim_matches('"')
+            .to_string();
+        let u_lifted_name = format!("{}_lifted", u_name);
+        let v_lifted_name = format!("{}_lifted", v_name);
+        let weight = weight_fn(&edge)?;
+        let gi_u_id = gi_name_to_node_index[&u_name];
+        let gi_v_id = gi_name_to_node_index[&v_name];
+        let gi_u = NodeIndex::new(gi_u_id);
+        let gi_v = NodeIndex::new(gi_v_id);
+        if orth.contains(&(u_index, v_index)) || orth.contains(&(v_index, u_index)) {
+            // Add cross edges with weight
+            gi.add_edge(
+                gi_u,
+                NodeIndex::new(gi_name_to_node_index[&v_lifted_name]),
+                weight,
+            );
+            gi.add_edge(
+                NodeIndex::new(gi_name_to_node_index[&u_lifted_name]),
+                gi_v,
+                weight,
+            );
+        } else {
+            // Add in-plane edges with weight
+            gi.add_edge(
+                NodeIndex::new(gi_name_to_node_index[&u_name]),
+                NodeIndex::new(gi_name_to_node_index[&v_name]),
+                weight,
+            );
+            gi.add_edge(
+                NodeIndex::new(gi_name_to_node_index[&u_lifted_name]),
+                NodeIndex::new(gi_name_to_node_index[&v_lifted_name]),
+                weight,
+            );
+        }
+    }
+    // Instead of finding the shortest path between each node and its lifted node, store the shortest paths in a list to find the shortest paths among them
+    let mut shortest_path_map: HashMap<String, i32> = HashMap::new();
+    for nodeid in graph.node_identifiers() {
+        let node_weight = graph.node_weight(nodeid).unwrap();
+        let node_name = format!("{:?}", node_weight).trim_matches('"').to_string();
+        let lifted_node_name = format!("{}_lifted", node_name);
+        let node = gi_name_to_node_index[&node_name];
+        let nodeidx = NodeIndex::new(node);
+        let lifted_node = gi_name_to_node_index[&lifted_node_name];
+        let lifted_nodeidx = NodeIndex::new(lifted_node);
+        let result = astar(
```

Comment:
Should probably just be a call for [dijkstra](https://docs.rs/rustworkx-core/latest/rustworkx_core/shortest_path/fn.dijkstra.html) given the A* heuristic is a constant